### PR TITLE
Add false to return types of class_implements()

### PR DESCRIPTION
**Pull request by melkamar**
The \class_implements() function may return an array or false when it fails

Based on https://github.com/phpstan/phpstan/issues/4172

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
Could you also update class_parents and class_uses?

**Comment by melkamar** [comment]
Done. Do I need to squash the commits or can you do that when accepting the PR?

**Comment by ondrejmirtes** [comment]
Now handling in PHPStan itself needs to be fixed :) It's fine, I can squash it before merging.

File: resources/functionMap.php [hunk 1]
```diff
@@ -949,9 +949,9 @@
 'chunk_split' => ['string', 'str'=>'string', 'chunklen='=>'int', 'ending='=>'string'],
 'class_alias' => ['bool', 'user_class_name'=>'string', 'alias_name'=>'string', 'autoload='=>'bool'],
 'class_exists' => ['bool', 'classname'=>'string', 'autoload='=>'bool'],
-'class_implements' => ['array<string,string>', 'what'=>'object|string', 'autoload='=>'bool'],
-'class_parents' => ['array<string,class-string>', 'instance'=>'object|string', 'autoload='=>'bool'],
-'class_uses' => ['array<string,string>', 'what'=>'object|string', 'autoload='=>'bool'],
+'class_implements' => ['array<string,string>|false', 'what'=>'object|string', 'autoload='=>'bool'],
```
Comment (staabm) [inline resources/functionMap.php:952]: is a interface also a `class-string` which can be used instead of `string` ?

Comment (ondrejmirtes) [reply resources/functionMap.php:952]: Yes

**Comment by ondrejmirtes** [comment]
Already done on master, but thanks.

**Pull request closed without merging**